validator/app/src/compute_horde_validator/validator/utils.py

```diff
@@ -0,0 +1,19 @@
+import time
+from collections.abc import Callable
+from datetime import timedelta
+
+
+class Timer:
+    def __init__(self, name: str = "", output_fn: Callable | None = None):
+        self.name = name
+        self.output_fn = output_fn
+
+    def __enter__(self):
+        self.start = time.time()
+        return self
+
+    def __exit__(self, *args):
+        self.end = time.time()
+        self.elapsed = timedelta(seconds=self.end - self.start)
+        if self.output_fn:
+            self.output_fn(f"{self.name} took {self.interval:.2f} seconds")
```

Comment:
```suggestion
            self.output_fn(f"{self.name} took {self.elapsed:.2f} seconds")
```

but IMO output_fn used this way doesn't seem versatile enough to be really useful - for example for logging you would like either structured log or at least %formatting 

maybe if it was just:
```suggestion
            self.output_fn(self)
```
then you could do whatever you wanted

Reply:
![download](https://github.com/backend-developers-ltd/ComputeHorde/assets/73276794/d753fe56-277a-4337-a474-b80bea5f5027)